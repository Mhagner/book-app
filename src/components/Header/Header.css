.header {
    display: flex;
    justify-content: space-between;
    padding: 2rem 0;
}

.header button {
    height: 3.8rem;
    background: transparent;
    border: none;
}

.header__logout-icon {
    margin-left: 2rem;
    transition: all 0.2s;
}

.header__logout-icon:hover {
    transform: scale(1.08)
}

.header__profile {
    display: flex;
    align-items: center;
    margin-bottom: 4.4rem;
}

.header__profile p {
    display: none;
    font-size: 1.2rem;
}

@media (min-width: 1026px) {
    .header__profile p {
        display: block;
    }
}
