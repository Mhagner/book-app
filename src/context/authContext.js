import React, { createContext, useContext } from 'react'

import { useAuth } from '../hooks/useAuth'

const AuthContext = createContext();

function AuthProvider({ children }) {
    const {
        handleSubmit,
        handleLogout,
        authenticate,
        loading,
        errors
    } = useAuth()

    return (
        <AuthContext.Provider value={{
            handleSubmit,
            handleLogout,
            authenticate,
            loading,
            errors
        }}>
            {children}
        </AuthContext.Provider>
    )

}

const useAuthContext = () => {
    return useContext(AuthContext);
}

export { useAuthContext, AuthProvider }