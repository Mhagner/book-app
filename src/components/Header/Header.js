import React from 'react'
import Logo from '../Logo/Logo'
import './Header.css'
import LogoutIcon from '../../assets/images/logout.svg'

const Header = ({ userName, handleLogout }) => {
    

    return (
        <div data-testid="header" className="header">    
            <Logo color="dark" />
            <div className="header__profile">
                {userName && <p>Bem vindo, <strong>{userName}!</strong></p>}
                <button onClick={handleLogout}>
                    <img
                        className="header__logout-icon"
                        src={LogoutIcon}
                        alt='logout'
                    />
                </button>
            </div>
        </div>
    )
}

export default Header