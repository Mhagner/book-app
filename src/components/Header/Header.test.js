import { render } from '@testing-library/react'
import { AuthProvider, useAuthContext } from '../../context/authContext'

import Header from './Header'

describe('Header component', () => {
    const renderComponent = () => {
        return render(<Header />)
    }

    it('Componente should defined', () => {
        const { getByTestId } = renderComponent()

        const element = getByTestId('header')

        expect(element).toBeDefined()
    })

    it('Componente should render', () => {

        const { getByTestId } = renderComponent()

        const element = getByTestId('header')

        expect(element).toMatchSnapshot()
    })

})