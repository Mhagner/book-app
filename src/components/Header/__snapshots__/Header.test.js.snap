// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Componente should render 1`] = `
<div
  class="header"
  data-testid="header"
>
  <div
    class="logo__container"
    data-testid="logo"
  >
    <img
      alt="ioasys"
      class="logo__image"
      src="logo-dark.png"
    />
    <span
      class="logo__text "
    >
      Books
    </span>
  </div>
  <div
    class="header__profile"
  >
    <button>
      <img
        alt="logout"
        class="header__logout-icon"
        src="logout.svg"
      />
    </button>
  </div>
</div>
`;
